src/main/java/com/aws/greengrass/shadowmanager/ShadowManagerDAOImpl.java

```diff
@@ -360,4 +366,14 @@ private <T> T execute(String sql, SQLExecution<T> thunk) {
             throw new ShadowManagerDataException(e);
         }
     }
+
+    private <T> T executeWriteOperation(String sql, SQLExecution<T> thunk) {
+        try {
+            return database.getDbWriteThreadPool().submit(() ->
+                    execute(sql, thunk)).get(10, TimeUnit.SECONDS);
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
```

Comment:
do we need to reset interrupt flag here? or if not i think a comment would be useful